### PR TITLE
[Add/Edit/View Observation] Support date and time fields

### LINKED ISSUE #748: [Add/edit observation] Add support for date and time field types
The observation screen should accept date and time input as well apart from text, multiple choice and photo.

### PR DESCRIPTION
**Pull request by HemanParbhakar**
Date & Time Input Support For Application.

Fixes #748 

@atalekar @gino-m PTAL?

### CODE REVIEW

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/DateFieldViewModel.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  @Hot(replays = true)
+  private final MutableLiveData<Nil> showDialogClicks = new MutableLiveData<>();
+
+  @Inject
+  DateFieldViewModel(Application application) {
+    super(application);
+  }
+
+  MutableLiveData<Nil> showDateDialog() {
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/ui/editobservation/DateFieldViewModel.java:35]: Also, in idiomatic Java the name of this method would imply that it causes the date dialog to be shown. Instead, please rename to `getDateDialogClicks()`

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java [hunk 2]
```diff
@@ -299,6 +327,14 @@ private void onSelectPhotoClick(int type, Field field) {
     }
   }
 
+  private void showDateDialog(){
+    //TODO: Implement Date Picker Dialog
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java:331]: Nit: As per Google Java style, space after "//", final ".".

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java [hunk 3]
```diff
@@ -299,6 +327,14 @@ private void onSelectPhotoClick(int type, Field field) {
...
+  }
+
+  private void showTimeDialog(){
+    //TODO: Implement Time Picker Dialog
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java:335]: Same here.

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/TimeFieldViewModel.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+  @Hot(replays = true)
+  private final MutableLiveData<Nil> showDialogClicks = new MutableLiveData<>();
+
+  @Inject
+  TimeFieldViewModel(Application application) {
+    super(application);
+  }
+
+  MutableLiveData<Nil> showTimeDialog() {
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/ui/editobservation/TimeFieldViewModel.java:35]: ->Observable<Nil>
->getShowTimeDialog()

File: gnd/src/main/res/layout/date_input_field.xml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        android:layout_height="wrap_content"
+        android:maxLines="1"
+        android:enabled="true"
+        android:inputType="datetime"
+        android:clickable="true"
+        android:focusableInTouchMode="false"
+        android:focusable="false"
+        android:text="@{viewModel.responseText}"
+        app:error="@{viewModel.error}"
+        tools:text="Date Time" />
```
Comment (gino-m) [inline gnd/src/main/res/layout/date_input_field.xml:52]: Please move text label to string resources instead.

Comment (gino-m) [inline gnd/src/main/res/layout/date_input_field.xml:52]: Also, should this say just "Date"?

**Review by gino-m — CHANGES_REQUESTED** [review]
Nice work! A few small suggestions. Please lmk when ready and I'll take another look!

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/DateFieldViewModel.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (HemanParbhakar) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/DateFieldViewModel.java]: replaced MutableLiveData<Nil> with Subject<Nil>

Comment (HemanParbhakar) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/DateFieldViewModel.java]: updated method name

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java (hunk 3 above)
Comment (HemanParbhakar) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java]: fixed

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/TimeFieldViewModel.java (hunk 4 above)
Comment (HemanParbhakar) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/TimeFieldViewModel.java]: replaced MutableLiveData with Subject and updated method name as well

File: gnd/src/main/res/layout/date_input_field.xml (hunk 5 above)
Comment (HemanParbhakar) [reply gnd/src/main/res/layout/date_input_field.xml]: moved to strings.xml

Comment (HemanParbhakar) [reply gnd/src/main/res/layout/date_input_field.xml]: changed and moved to strings.xml

**Comment by HemanParbhakar** [comment]
@gino-m @atalekar this PR can be ignored latest PR is #846

**Pull request closed without merging**